pwiz_tools/Skyline/Controls/Graphs/GraphFullScan.cs

```diff
@@ -598,13 +604,125 @@ private void PopulateProperties()
                     var chromSet = stateProvider.DocumentUI.Settings.MeasuredResults.Chromatograms.FirstOrDefault(
                         chrom => chrom.ContainsFile(_msDataFileScanHelper.ScanProvider.DataFilePath));
                     spectrumProperties.ReplicateName = chromSet?.Name;
+                    if (_peaks != null && _peaks.Length > 0)
+                    {
+                        var nodePath = DocNodePath.GetNodePath(_msDataFileScanHelper.CurrentTransition?.Id,
+                            _documentContainer.DocumentUI);
+
+                        if (!nodePath.Precursor.Id.Equals(_precursor?.DocNode?.Id))
+                            _precursor = new GraphSpectrum.Precursor(_documentContainer.DocumentUI.Settings, null,
+                                nodePath.Peptide, nodePath.Precursor);
+
+                        if (_msDataFileScanHelper.Source == ChromSource.fragment) // Calculate library dotp
+                        {
+                            if (_precursor.Spectra != null && _precursor.Spectra.Count > 0)
+                            {
+                                var libSpectrum = _precursor.Spectra[0].SpectrumPeaksInfo;
+
+                                // Rank the library spectrum with the most generic options. We need it to identify ion types

```

Comment:
This is interesting.
I was imagining that you would calculate the dot product between the raw file spectrum and the library spectrum only based on the m/z's of the TransitionDocNode's that are in the document.
Instead, it looks like you are going to be calculating the dot product using all of the m/z's that could be identified as a particular fragment.

I think it is better to only look at the m/z's that are in the document because the user has told us m/z's they care about. They might have deleted a transition from their document because they knew it had interference and so they would not want it included in any dot product.

Reply:
I guess you didn't look through all the changes. I do use document transitions only. I use the SpectrumRanker to mark up the library peaks because I want to make sure that the peaks used for dotp calculation are same as the ones used for the area dotp. Also there is a possibility to add another dotp value for the full set of annotated peaks and the marked up spectrum will be useful there.

Reply:
Oh. I think I understand now.

The TransitionDocNode's know what their intensities from the library spectrum are. It's stored in "TransitionDocNode.LibInfo.Intensity".
It might make more sense to use the TransitionDocNode.LibInfo.Intensity values to calculate the dot product, instead of reading the library spectrum from the .blib file again.